File: Serene/Serene.Script/Imports/FormContexts/Northwind.RegionForm.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class RegionForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Region";
    
        public RegionForm(string idPrefix) : base(idPrefix) {}
    
        public IntegerEditor RegionID { [InlineCode("{this}.w('RegionID', Serenity.IntegerEditor)")] get; private set; }
        public StringEditor RegionDescription { [InlineCode("{this}.w('RegionDescription', Serenity.StringEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Imports/ServerTypings/Northwind.RegionForm.ts
namespace Serene.Northwind {
    export interface RegionForm {
        RegionID: Serenity.IntegerEditor;
        RegionDescription: Serenity.StringEditor;
    }

    export class RegionForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Region';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!RegionForm.init)  {
                RegionForm.init = true;

                var s = Serenity;
                var w0 = s.IntegerEditor;
                var w1 = s.StringEditor;

                Q.initFormType(RegionForm, [
                    'RegionID', w0,
                    'RegionDescription', w1
                ]);
            }
        }
    }
}

